.github/workflows/ci-cd.yml

```diff
@@ -17,8 +17,46 @@ env:
   POSTGRES_PASSWORD: postgres
 
 jobs:
+  bundle-install:
```

Comment:
`bundle-install` and `yarn-install` step exist as the first parallel step because it can download and cache new dependencies for the rest of the build to use. 

`yarn-install` is more special because it does a "lookup only" cache fetch to verify if the file exist first.